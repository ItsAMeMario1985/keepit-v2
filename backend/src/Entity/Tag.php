<?php

namespace App\Entity;

use App\Repository\TagRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass=TagRepository::class)
 */
class Tag
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $value;

    /**
     * @ORM\ManyToOne(targetEntity=Image::class, inversedBy="tags")
     */
    private $image;

    /**
     * @ORM\ManyToOne(targetEntity=User::class, inversedBy="tags")
     */
    private $user;

    /**
     * @ORM\ManyToMany(targetEntity=Keepit::class, inversedBy="tags",cascade={"persist"})
     */
    private $keepit;

    /**
     * @ORM\Column(type="boolean", nullable=true)
     */
    private $isCustom;

    public function __construct()
    {
        $this->keepit = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getValue(): ?string
    {
        return $this->value;
    }

    public function setValue(string $value): self
    {
        $this->value = $value;

        return $this;
    }

    public function getImage(): ?Image
    {
        return $this->image;
    }

    public function setImage(?Image $image): self
    {
        $this->image = $image;

        return $this;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): self
    {
        $this->user = $user;

        return $this;
    }

    /**
     * @return Collection|Keepit[]
     */
    public function getKeepit(): Collection
    {
        return $this->keepit;
    }

    public function addKeepit(Keepit $keepit): self
    {
        if (!$this->keepit->contains($keepit)) {
            $this->keepit[] = $keepit;
        }

        return $this;
    }

    public function removeKeepit(Keepit $keepit): self
    {
        $this->keepit->removeElement($keepit);

        return $this;
    }

    public function getIsCustom(): ?bool
    {
        return $this->isCustom;
    }

    public function setIsCustom(?bool $isCustom): self
    {
        $this->isCustom = $isCustom;

        return $this;
    }
}
