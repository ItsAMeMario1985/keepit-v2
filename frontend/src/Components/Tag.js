import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import { ReactComponent as TagCloudIcon } from '../Assets/tag-cloud.svg'
import { ReactComponent as TagMinusIcon } from '../Assets/tag-minus.svg'
import { ReactComponent as TagPlusIcon } from '../Assets/tag-plus.svg'
import { ReactComponent as TagUserIcon } from '../Assets/tag-user.svg'

export default function Tag({ onClick, tagValue, targetState }) {
  const stateIcon = targetState ? <TagPlusIcon /> : <TagMinusIcon />
  const bgColor = targetState
    ? 'var(--color-primary)'
    : 'var(--color-secondary)'

  function IsCustomIcon(test) {
    if (test.isCustom) {
      return <TagUserIcon />
    } else {
      return <TagCloudIcon />
    }
  }

  return (
    <StyledTag bgcolor={bgColor} onClick={() => onClick(tagValue, targetState)}>
      {tagValue}
    </StyledTag>
  )
}

const StyledTag = styled.span`
  background-color: ${(props) => props.bgcolor};
  display: inline-block;
  border-bottom-right-radius: 8px;
  border-top-right-radius: 8px;

  display: inline-flex;
  align-items: center;
  padding: 5px;
  margin: 0 7px 7px 0px;
  color: white;
  font-weight: 400;
  font-size: 13px;
  letter-spacing: 0.5px;
`

/*

  background-color: ${(props) => props.bgcolor};
  display: inline-block;
  border-bottom-right-radius: 8px;
  border-top-right-radius: 8px;
  min-width: 50px;
  display: inline-flex;
  align-items: center;
  padding: 5px;
  margin: 0 7px 7px 0px;


*/
