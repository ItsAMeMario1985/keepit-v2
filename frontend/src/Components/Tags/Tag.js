import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

Tag.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  tagValue: PropTypes.string.isRequired,
  targetState: PropTypes.bool.isRequired,
}

export default function Tag({ onClick, tagValue, targetState }) {
  const bgColor = targetState
    ? 'var(--color-primary)'
    : 'var(--color-secondary)'

  return (
    <StyledTag bgcolor={bgColor} onClick={() => onClick(tagValue, targetState)}>
      {tagValue}
    </StyledTag>
  )
}

const StyledTag = styled.span`
  background-color: ${(props) => props.bgcolor};
  display: inline-block;
  border-bottom-right-radius: 8px;
  border-top-right-radius: 8px;
  display: inline-flex;
  align-items: center;
  padding: 5px;
  margin: 0 7px 7px 0px;
  color: white;
  font-weight: 400;
  font-size: 13px;
  letter-spacing: 0.5px;
`
