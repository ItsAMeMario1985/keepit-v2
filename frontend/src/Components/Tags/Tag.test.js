import { render } from '@testing-library/react'
import Tag from './Tag'
describe('Tag', () => {
  it('renders correctly', () => {
    const { container } = render(<Tag />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
