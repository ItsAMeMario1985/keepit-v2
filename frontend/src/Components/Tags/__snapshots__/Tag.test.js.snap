// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag renders correctly 1`] = `
<span
  class="Tag__StyledTag-sc-1f6t5zy-0 flnIy"
>
  Test
</span>
`;
